clowdapp.yaml

```diff
@@ -0,0 +1,190 @@
+            initialDelaySeconds: 10
+            periodSeconds: 10
+        resources:
+          limits:
+            cpu: ${CPU_LIMIT}
+            memory: {MEMORY_LIMIT}
+          requests:
+            cpu: ${CPU_REQUEST}
+            memory: ${MEMORY_REQUEST}
+        volumes:
+          - emptyDir: {}
+            name: system-baseline-prometheus-data
+        volumeMounts:
+          - mountPath: /prometheus-data
+            name: system-baseline-prometheus-data
+        env:
+          - name: APP_CONFIG
+            value: gunicorn.conf.py
+          - name: ENABLE_RBAC
+            value: "${DRIFT_ENABLE_RBAC}"
+          - name: ENABLE_SMART_MANGEMENT_ENTITLEMENT_CHECK
+            value: "${ENABLE_SMART_MANGEMENT_ENTITLEMENT_CHECK}"
+          - name: WEB_CONCURRENCY
+            value: "4"
+          - name: PATH_PREFIX
+            value: /api/
+          - name: APP_NAME
+            value: system-baseline
+          - name: ENABLE_PIPENV
+            value: 'true'
+          - name: prometheus_multiproc_dir
+            value: /prometheus-data
+          - name: DRIFT_SHARED_SECRET
+            valueFrom:
+              secretKeyRef:
+                key: shared-secret
+                name: drift-interpod-comms
+
+- apiVersion: v1
+  kind: Service
+  metadata:
+    labels:
+      app: system-baseline
+    name: system-baseline
+    annotations:
+      prometheus.io/path: /mgmt/v0/metrics
+      prometheus.io/port: '8000'
+      prometheus.io/scrape: 'true'
+  spec:
+    type: ClusterIP
+    selector:
+      deploymentconfig: system-baseline
+    ports:
+      - name: 8000-tcp
+        port: 8000
+        protocol: TCP
+        targetPort: 8000
+
+- apiVersion: autoscaling/v1
+  kind: HorizontalPodAutoscaler
+  metadata:
+    name: system-baseline
+    labels:
+      app: system-baseline
+  spec:
+    scaleTargetRef:
+      apiVersion: v1
+      kind: DeploymentConfig
+      name: system-baseline
+    minReplicas: ${{MIN_REPLICAS}}
+    maxReplicas: ${{MAX_REPLICAS}}
+    targetCPUUtilizationPercentage: 50
+
+parameters:
+  - name: DRIFT_ENABLE_RBAC
+    description: enable rbac. needs to be "True" or "False".
+    value: "True"
+  - name: ENABLE_SMART_MANGEMENT_ENTITLEMENT_CHECK
+    description: enable the smart mgmt entitlement check on requests. Needs to be "True" or "False".
+    value: "False"
+  - name: IMAGE_TAG
+    description: image tag to pull
+    value: latest
+  - name: IMAGE
+    description: image name
+    value: quay.io/cloudservices/system-baseline-backend
+  - name: MIN_REPLICAS
+    description: minimum pod replicas for horizontal pod autoscaler
+    value: "1"
+  - name: MAX_REPLICAS
+    description: maximum pod replicas for horizontal pod autoscaler
+    value: "1"
+  - name: CPU_LIMIT
+    value: 500m
+  - name: MEMORY_LIMIT
+    value: 500Mi
+  - name: CPU_REQUEST
+    value: 100m
+  - name: MEMORY_REQUEST
+    value: 300Mi
+  - description: ClowdEnv Name
+    name: ENV_NAME
+  - name: CLOWDER_ENABLED
```

Comment:
Is this used somewhere? Or needed for clowder to work?

Reply:
ENV_NAME is automagically inserted by Clowder as well, so the clowdapp can reference the correct ClowdEnvironment.

Please, look at: https://clouddot.pages.redhat.com/clowder/dev/migration/migration.html#_create_deployment_template_with_clowdapp_resource

Reply:
Awesome!